frontend/app/routes/protected/renew/$id/review-adult-information.tsx

```diff
@@ -59,64 +59,89 @@ export async function loader({ context: { appContainer, session }, params, reque
+    address: state.addressInformation?.mailingAddress ?? state.clientApplication.contactInformation.mailingAddress,
+    city: state.addressInformation?.mailingCity ?? state.clientApplication.contactInformation.mailingCity,
+    province: mailingProvinceTerritoryStateAbbr,
+    postalCode: state.addressInformation?.mailingPostalCode ?? state.clientApplication.contactInformation.mailingPostalCode,
+    country: mailingCountryAbbr,
+    apartment: state.addressInformation?.mailingApartment ?? state.clientApplication.contactInformation.mailingApartment,
   };
 
   const homeAddressInfo = {
-    address: state.clientApplication.contactInformation.homeAddress,
-    city: state.clientApplication.contactInformation.homeCity,
-    province: state.clientApplication.contactInformation.homeProvince,
-    postalCode: state.clientApplication.contactInformation.homePostalCode,
-    country: state.clientApplication.contactInformation.homeCountry,
-    apartment: state.clientApplication.contactInformation.homeApartment,
+    address: state.addressInformation?.homeAddress ?? state.clientApplication.contactInformation.homeAddress,
+    city: state.addressInformation?.homeCity ?? state.clientApplication.contactInformation.homeCity,
+    province: homeProvinceTerritoryStateAbbr,
+    postalCode: state.addressInformation?.homePostalCode ?? state.clientApplication.contactInformation.homePostalCode,
+    country: homeCountryAbbr,
+    apartment: state.addressInformation?.homeApartment ?? state.clientApplication.contactInformation.homeApartment,
   };
 
   const dentalInsurance = state.dentalInsurance;
 
-  const selectedFederalGovernmentInsurancePlan = state.dentalBenefits.federalSocialProgram
+  const selectedFederalGovernmentInsurancePlan = state.dentalBenefits?.federalSocialProgram
     ? appContainer.get(TYPES.domain.services.FederalGovernmentInsurancePlanService).getLocalizedFederalGovernmentInsurancePlanById(state.dentalBenefits.federalSocialProgram, locale)
     : undefined;
 
-  const selectedProvincialBenefit = state.dentalBenefits.provincialTerritorialSocialProgram
+  const selectedProvincialBenefit = state.dentalBenefits?.provincialTerritorialSocialProgram
     ? appContainer.get(TYPES.domain.services.ProvincialGovernmentInsurancePlanService).getLocalizedProvincialGovernmentInsurancePlanById(state.dentalBenefits.provincialTerritorialSocialProgram, locale)
     : undefined;
 
   const dentalBenefit = {
     federalBenefit: {
-      access: state.dentalBenefits.hasFederalBenefits,
+      access: state.dentalBenefits?.hasFederalBenefits,
       benefit: selectedFederalGovernmentInsurancePlan?.name,
     },
     provTerrBenefit: {
-      access: state.dentalBenefits.hasProvincialTerritorialBenefits,
-      province: state.dentalBenefits.province,
+      access: state.dentalBenefits?.hasProvincialTerritorialBenefits,
+      province: state.dentalBenefits?.province,
       benefit: selectedProvincialBenefit?.name,
     },
```

Comment:
We should default to the `clientApplication.dentalBenefits` if `state.dentalBenefits` is undefined.

I feel like we can simplify the structure of the `dentalBenefit` local variable.

Since `clientApplication.dentalBenefits`, which holds the client info on file, is an array, we could make this `dentalBenefit` local variable an array as well and populate it with the user's selected federal and provincial/territorial beneifts if they have access to them. 

I don't see a need to pass the `province` value to the front-end - it doesn't look like it's used.